File: api/src/Entity/Token.php
<?php

namespace App\Entity;

use DateTime;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity()
 * @ORM\Table(name="tokens", uniqueConstraints={@ORM\UniqueConstraint(name="tokens_value_unique", columns={"token_value"})})
 */
class Token
{
    /**
     * @ORM\Id
     * @ORM\Column(type="string")
     * @ORM\GeneratedValue(strategy="UUID")
     */
    protected $token_id;

    /**
     * @ORM\Column(type="string")
     */
    protected $token_value;

    /**
     * @ORM\Column(type="datetime")
     * @var DateTime
     */
    protected $token_created_at;

    /**
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumn(name="token_user", referencedColumnName="user_id", onDelete="CASCADE")
     * @var User
     */
    protected $token_user;

    public function getTokenId()
    {
        return $this->token_id;
    }

    public function setTokenId($id)
    {
        $this->token_id = $id;
        return $this;
    }

    public function getTokenValue()
    {
        return $this->token_value;
    }

    public function setTokenValue($value)
    {
        $this->token_value = $value;
        return $this;
    }

    public function getTokenCreatedAt()
    {
        return $this->token_created_at;
    }

    public function setTokenCreatedAt(DateTime $createdAt)
    {
        $this->token_created_at = $createdAt;
        return $this;
    }

    public function getTokenUser()
    {
        return $this->token_user;
    }

    public function setTokenUser(User $user)
    {
        $this->token_user = $user;
        return $this;
    }
}

File: openex-model/src/main/java/io/openex/database/model/Token.java
package io.openex.database.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.openex.helper.MonoIdDeserializer;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.time.Instant;
import java.util.Objects;

@Entity
@Table(name = "tokens")
public class Token implements Base {
    @Id
    @Column(name = "token_id")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @JsonProperty("token_id")
    private String id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "token_user")
    @JsonSerialize(using = MonoIdDeserializer.class)
    @JsonProperty("token_user")
    private User user;

    @Column(name = "token_value")
    @JsonProperty("token_value")
    private String value;

    @Column(name = "token_created_at")
    @JsonProperty("token_created_at")
    private Instant created;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public Instant getCreated() {
        return created;
    }

    public void setCreated(Instant created) {
        this.created = created;
    }

    @Override
    public boolean isUserHasAccess(User user) {
        return this.user.equals(user);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || !Base.class.isAssignableFrom(o.getClass())) return false;
        Base base = (Base) o;
        return id.equals(base.getId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}
